roles/openshift_facts/library/openshift_facts.py

```diff
@@ -430,6 +408,48 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_deployment_facts_if_unset(facts):
+    """ Set Facts that vary based on deployment_type. This currently
+        includes common.service_type, common.config_base, master.registry_url,
+        node.registry_url
+
+        Args:
+            facts (dict): existing facts
+        Returns:
+            dict: the facts dict updated with the generated deployment_type
+            facts
+    """
+    if 'common' in facts:
+        deployment_type = facts['common']['deployment_type']
+        if 'service_type' not in facts['common']:
+            service_type = 'atomic-openshift'
+            if deployment_type == 'origin':
+                service_type = 'openshift'
+            elif deployment_type in ['enterprise', 'online']:
+                service_type = 'openshift'
+            facts['common']['service_type'] = service_type
+        if 'config_base' not in facts['common']:
+            config_base = '/etc/origin'
+            if deployment_type in ['enterprise', 'online']:
+                config_base = '/etc/openshift'
+            elif deployment_type == 'origin':
+                config_base = '/etc/openshift'
+            facts['common']['config_base'] = config_base
+
+    for role in ('master', 'node'):
+        if role in facts:
+            deployment_type = facts['common']['deployment_type']
+            if 'registry_url' not in facts[role]:
+                registry_url = 'aos3/aos-${component}:${version}'
+                if deployment_type in ['enterprise', 'online']:
+                    registry_url = 'openshift3/ose-${component}:${version}'
```

Comment:
Though I suspect online overrides this to reference a custom registry, the default would remain `openshift3/ose-${component}:${version}`